minetest/minetest
# [NOSQUASH] Fix out-of-bounds access in NodeDefManager::nodeboxConnects, and add more connected nodebox testnodes

Fixes #13377. (See last commit.)

The other commits are just code reformatting, so that the array is more readable, and testnodes for devtest.

Related: Properly fixing the connection stuff for arbitrary facedir rotations would be #7696.

## To do

This PR is a Ready for Review.

## How to test

* Compile with sanitizers.
* Place a `testnodes:facedir_to_connect_to` node, and place connected nodebox nodes (e.g. `testnodes:nodebox_connected_6side`) around it.
* Rotate the `facedir_to_connect_to` with the param2 tool.
* Notice how the sanitizers cause an abort without the last commit.
